### PR TITLE
Implement new API tp.is_deletable

### PR DESCRIPTION
**Pull request by jrobinAV**
#617 
- Must update the documentation + release notes
- Must propagate the change on __init__.py file to other repositories

### CODE REVIEW

File: src/taipy/__init__.py
Comment (trgiangdo) [inline src/taipy/__init__.py:1]: Why are we removing some imports from this file?

File: src/taipy/core/scenario/_scenario_manager.py [hunk 1]
```diff
@@ -280,6 +268,27 @@ def _compare(cls, *scenarios: Scenario, data_node_config_id: Optional[str] = Non
...
+        if scenario.is_primary:
+            if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
+                return False
+        return True
+
+    @classmethod
+    def _delete(cls, scenario_id: ScenarioId):
+        scenario = cls._get(scenario_id)
+        if scenario.is_primary:
+            if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
```
Comment (trgiangdo) [inline src/taipy/core/scenario/_scenario_manager.py]: Maybe we can refactor this part, something like if not _is_deletable then raise error.
And the `_hard_delete()` method as well.
What do you think?

**Review by trgiangdo — COMMENTED** [review]
I have a question and a minor comment

File: src/taipy/__init__.py
Comment (jrobinAV) [reply src/taipy/__init__.py:1]: Oh... Mypy did it cause they are not used. 
We must not delete the imports. I will revert this part.
Thx!

Comment (jrobinAV) [reply src/taipy/__init__.py:1]: Even more, we need to add the new method is_deletable to the imports and propagate it to other repos.

File: src/taipy/core/scenario/_scenario_manager.py (hunk 1 above)
Comment (jrobinAV) [reply src/taipy/core/scenario/_scenario_manager.py]: Yes, good idea.

**Review by trgiangdo — APPROVED** [review]
Neat!

**Review by toan-quach — APPROVED** [review]
lgtm :D

**Pull request merged**